packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -285,10 +320,21 @@ export class SCWSigner implements Signer {
         }
 
         const spendLimits = response?.accounts?.[0].capabilities?.spendLimits;
+
         if (spendLimits && 'permissions' in spendLimits) {
-          store.spendLimits.set({
-            [this.chain.id]: spendLimits.permissions,
-          });
+          const spendLimitsByChainId = spendLimits?.permissions.reduce(
+            (acc, permission) => {
+              if (!acc[permission.chainId]) {
+                acc[permission.chainId] = [];
+              }
+
+              acc[permission.chainId].push(permission);
+              return acc;
+            },
+            {} as Record<number, SpendLimit[]>
+          );
+
+          store.spendLimits.set(spendLimitsByChainId);
```

Comment:
if we're okay with just storing the spend limits array as it comes from the response instead of storing it by chain ID we can remove this step

Reply:
yeah makes sense. will update the storage type and remove this